src/coroutine/memoset/prove.rs

```diff
@@ -0,0 +1,354 @@
+use crate::{
+    circuit::gadgets::pointer::AllocatedPtr,
+    error::ProofError,
+    field::LurkField,
+    lem::{pointers::Ptr, store::Store},
+    proof::{
+        nova::{CurveCycleEquipped, E1},
+        supernova::{Proof, PublicParams, SuperNovaPublicParams, C2, SS1, SS2},
+        FrameLike, Prover, RecursiveSNARKTrait,
+    },
+};
+
+use bellpepper_core::{num::AllocatedNum, ConstraintSystem, SynthesisError};
+use once_cell::sync::OnceCell;
+use tracing::info;
+
+use super::query::Query;
+use super::{
+    CoroutineCircuit, LogMemo, MemoSet, Scope, DEFAULT_RC_FOR_QUERY,
+    DEFAULT_TRANSCRIBE_INTERNAL_INSERTIONS,
+};
+
+use nova::{
+    supernova::{
+        error::SuperNovaError, snark::CompressedSNARK, NonUniformCircuit, RecursiveSNARK,
+        StepCircuit,
+    },
+    traits::{
+        snark::{BatchedRelaxedR1CSSNARKTrait, RelaxedR1CSSNARKTrait},
+        Dual as DualEng,
+    },
+};
+use std::marker::PhantomData;
+
+const COROUTINE_ARITY: usize = 12;
```

Comment:
Can you document what this represents? My guess is that this correspond to the CEK IO, but that doesn't account for the transcript and acc (new IO), so I'm not quite sure.

Reply:
It represents the input size (or the output size). 6 for CEK, 6 for memoset stuff

Reply:
Aha! Thanks.